src/VisualStudio/VisualBasic/Impl/ProjectSystemShim/EntryPointFinder.vb

```diff
@@ -30,7 +31,12 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.ProjectSystemShim
         End Function
 
         Public Overrides Sub VisitNamedType(symbol As INamedTypeSymbol)
-            ' It's a form if it Inherits System.Windows.Forms.Form. 
+            ' Metadata symbols cannot be entry points.
+            If Not symbol.IsFromSource Then
```

Comment:
If we only want to walk source, an easier way is to just get the source assembly namespace for a project and walk down that. Then you skip metadata symbols entirely.

Reply:
yeah this was a lot simpler, thanks!